packages/dd-trace/src/llmobs/writers/util.js

```diff
@@ -0,0 +1,60 @@
+'use strict'
+
+const logger = require('../../log')
+const { EVP_PROXY_AGENT_BASE_PATH } = require('../constants/writers')
+const telemetry = require('../telemetry')
+
+const AgentInfoExporter = require('../../exporters/common/agent-info-exporter')
+/** @type {AgentInfoExporter} */
+let agentInfoExporter
+
+function setAgentStrategy (config, setWritersAgentlessValue) {
+  const agentlessEnabled = config.llmobs.agentlessEnabled
+
+  if (agentlessEnabled != null) {
+    setWritersAgentlessValue(agentlessEnabled)
+    return
+  }
+
+  if (!agentInfoExporter) {
+    agentInfoExporter = new AgentInfoExporter(config)
+  }
+
+  agentInfoExporter.getAgentInfo((err, agentInfo) => {
+    if (err) {
+      setWritersAgentlessValue(true)
+      return
+    }
+
+    const endpoints = agentInfo.endpoints
+    const hasEndpoint = Array.isArray(endpoints) && endpoints.some(endpoint => endpoint === EVP_PROXY_AGENT_BASE_PATH)
+    setWritersAgentlessValue(!hasEndpoint)
+  })
+}
+
+function parseResponseAndLog (err, code, eventsLength, url, eventType) {
+  if (code === 403 && err.message.includes('API key is invalid')) {
```

Comment:
i've decided to special-case this here to make the message a bit clearer/more actionable, and still send the telemetry event as `request_error`. I have telemetry submission here too but now that i think about it, if the api key and site don't match, and they're likely using agentless telemetry too, i doubt the telemetry event would even make it to our end...